<?xml version="1.0" encoding="utf-8"?>
<ui version="4.0">
	<class>MainWindow</class>
	<widget class="QMainWindow" name="MainWindow">
		<property name="geometry">
			<rect>
				<x>0</x>
				<y>0</y>
				<width>800</width>
				<height>600</height>
			</rect>
		</property>
		<property name="windowTitle">
			<string>Haushaltsbuch</string>
		</property>
		<widget class="QWidget" name="centralWidget">
			<layout class="QVBoxLayout" name="verticalLayout">
				
				<!-- Input Section -->
				<item>
					<widget class="QGroupBox" name="groupBox">
						<property name="title">
							<string>Neue Transaktion</string>						
						</property>
						<layout class="QGridLayout" name="gridLayout">
							<item row="0" column="0">
								<widget class="QLabel" name="label">
									<property name="text">
										<string>Datum:</string>
									</property>
								</widget>
							</item>
							<item row="0" column="1">
								<widget class="QDateEdit" name="dateEdit">
									<property name="date">
										<date>2025-01-01</date>
									</property>
									<property name="calendarPopup">
										<bool>true</bool>
									</property>
								</widget>
							</item>
							<item row="1" column="0">
								<widget class="QLabel" name="label_2">
									<property name="text">
										<string>Beschreibung:</string>
									</property>
								</widget>
							</item>
							<item row="1" column="1">
								<widget class="QLineEdit" name="descriptionEdit"/>
							</item>
							<item row="2" column="0">
								<widget class="QLabel" name="label_3">
									<property name="text">
										<string>Kategorie:</string>
									</property>
								</widget>
							</item>
							<item row="2" column="1">
								<widget class="QComboBox" name="categoryCombo"/>
							</item>
							<item row="3" column="0">
								<widget class="QLabel" name="label_4">
									<property name="text">
										<string>Betrag (€):</string>
									</property>
								</widget>
							</item>
							<item row="3" column="1">
								<widget class="QDoubleSpinBox" name="amountSpin">
									<property name="minimum">
										<double>0.000000000000000</double>
									</property>
									<property name="maximum">
										<double>1000000.000000000000000</double>
									</property>
									<property name="prefix">
										<string>€ </string>
									</property>
								</widget>
							</item>
							<item row="4" column="0">
								<widget class="QLabel" name="label_5">
									<property name="text">
										<string>Typ:</string>
									</property>
								</widget>
							</item>
							<item row="4" column="1">
								<widget class="QComboBox" name="typeCombo"/>
							</item>
						</layout>
					</widget>
				</item>
				
				<!-- Buttons -->
				<item>
					<layout class="QHBoxLayout" name="horizontalLayout">
						<item>
							<widget class="QPushButton" name="addButton">
								<property name="text">
									<string>Hinzufügen</string>
								</property>
							</widget>
						</item>
						<item>
							<widget class="QPushButton" name="editButton">
								<property name="text">
									<string>Bearbeiten</string>
								</property>
							</widget>
						</item>
						<item>
							<widget class="QPushButton" name="deleteButton">
								<property name="text">
									<string>Löschen</string>
								</property>
							</widget>
						</item>
						<item>
							<widget class="QPushButton" name="generatePushButton">
								<property name="text">
									<string>Bericht</string>
								</property>
							</widget>
						</item>
					</layout>
				</item>
				
				<!-- Search Section -->
				<item>
					<layout class="QHBoxLayout" name="horizontalLayout_2">
						<item>
							<widget class="QLineEdit" name="searchEdit">
								<property name="placeholderText">
									<string>Suche nach Beschreibung...</string>
								</property>
							</widget>
						</item>
						<item>
							<widget class ="QPushButton" name="searchButton">
								<property name ="text">
									<string>Suchen</string>
								</property>
							</widget>
						</item>
					</layout>
				</item>
				
				<!-- Transkationen -->
				<item>
					<widget class="QTableView" name="transactionTable">
						<property name="selectionMode">
							<enum>QAbstractItemView::SingleSelection</enum>
						</property>
						<property name="selectionBehavior">
							<enum>QAbstractItemView::SelectRows</enum>
						</property>
					</widget>
				</item>
				
				<!-- Summary Section -->
				<item>
					<layout class="QHBoxLayout" name="horizontalLayout_3">
						<item>
							<widget class="QLabel" name="incomeLabel">
								<property name="text">
									<string>Einnahmen: 0.00 €</string>
								</property>
							</widget>
						</item>
						<item>
							<widget class="QLabel" name="expensesLabel">
								<property name="text">
									<string>Ausgaben: 0.00 €</string>
								</property>
							</widget>
						</item>
						<item>
							<widget class="QLabel" name="balanceLabel">
								<property name="text">
									<string>Saldo: 0.00 €</string>
								</property>
							</widget>
						</item>
					</layout>
				</item>
						
			</layout>
		</widget>
	</widget>
	<resources/>
	<connections/>
</ui>